wordpress-mobile/WordPress-Android
# Migrate old Quick Start UI code to ImprovedMySiteFragment as Quick Start Block

Issue #15127

This PR migrates old `Quick Start` UI code from `MySiteFragment` to `ImprovedMySiteFragment` as a `Quick Start Block`.

To test:

1. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
2. Disable `QuickStartDynamicCardsFeatureConfig`, enable `MySiteImprovementsFeatureConfig`
3. Restart the app
4. Select a site with quick start in progress 
5. Notice that the old quick start section gets displayed according to completed/ uncompleted steps

![qs_improved_my_site](https://user-images.githubusercontent.com/1405144/128358824-8cf0b450-dfdf-482e-8097-be2b14607de3.png)

## Regression Notes
1. Potential unintended areas of impact
    - `ImprovedMySiteFragment` -> `Quick Start - Dynamic Card` (`MySiteImprovementsFeatureConfig` - ON, `QuickStartDynamicCardsFeatureConfig` - ON )
    - `MySiteFragment` -> `Quick Start` (`MySiteImprovementsFeatureConfig` - OFF)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for `Quick Start Block` display.

PR submission checklist:
